### PR TITLE
An assortment of changes done over Summer 2016 

### PR DESCRIPTION
**Pull request by vishaalprasad**

### CODE REVIEW

**Comment by vishaalprasad** [comment]
These are changes I made over summer. I'm not sure where the conflict is (perhaps in VisualizeDataset... I'm not quite sure why I committed those changes to begin with?) but most of it should speak for itself.

If anything is unclear, do let me know.

File: code/_expts/slotnick_etal_2001/de_StimCreate.m [hunk 1]
```diff
@@ -91,21 +91,47 @@
...
-            train.XLAB{end+1} = sprintf('(Dist) Left: %dpx ; Right: %dpx', d1, d2);
+            train.XLAB{end+1} = sprintf('(Dist) Top: %dpx ; Bottom: %dpx', d1, d2);
 
             % Coordinate task
             train.T(end+1) = (d1 == d2); % same distance or no?
             difficulty = guru_iff(d1 == d2, 'easy', 'hard');  % according to the paper...
             lbl = guru_iff(train.T(end) == 1, 'same', 'different');
             train.TLAB{end+1} = sprintf('%s-%s', lbl, difficulty);
+            
+            if d1 == d2
```
Comment (bcipolli) [inline code/_expts/slotnick_etal_2001/de_StimCreate.m:114]: I'm guessing this is the insertion to duplicate training examples. If so, I think the clearer way is to do:

```
train.X(:, end+1) = train.X(:, end);
train.XLAB{end+1} = train.XLAB{end}
```

Either way, please add a comment to this block about what you did and why.

File: code/analyze/ac/plots/de_PlotFFTs.m [hunk 2]
```diff
@@ -309,18 +309,30 @@
           pwrLH = squeeze(ffts.orig2D(1,1,:,:)).*(pd<0).*(ffts.pals.an2D<=0.05);
 
           imgRH = ifft2(pwrRH); imgRH=imgRH(1:mSets.nInput(1), 1:mSets.nInput(2));
+          rMin = min(min(imgRH));
```
Comment (bcipolli) [inline code/analyze/ac/plots/de_PlotFFTs.m]: I find `min(imgRH(:))` is a bit clearer, as it says: get the min over all dimensions. Your idiom reads: get the min over the first two dimensions; I have to know that there are only two.

File: code/analyze/ac/plots/de_PlotFFTs.m [hunk 3]
```diff
@@ -309,18 +309,30 @@
...
 
+
+          % Now normalize the images so they are properly visible
           % Now take an image, and show.
+         
 
           subplot(1,2,1); colormap gray;
           title(sprintf('LH (\\sigma=%3.2f) p<0.05 recon', mSets.sigma(end)));
           imagesc(imgLH, [0 1]);
+          ylabel('Normalized')
```
Comment (bcipolli) [inline code/analyze/ac/plots/de_PlotFFTs.m]: `Normalized` is an adjective. Perhaps `Normalized intensity`? Or `Normalized power`?

File: code/analyze/ac/plots/de_PlotFFTs.m (hunk 2 above)
Comment (bcipolli) [reply code/analyze/ac/plots/de_PlotFFTs.m]: You actually use that idiom below. Better to use a single idiom, so highly encourage the change here.

File: code/stim/de_VisualizeDataset.m [hunk 4]
```diff
@@ -7,19 +7,34 @@
...
   n_pix = prod(dset.nInput);
   img_shape = dset.nInput;
-  im2show     = de_SelectImages(dset, nImages);
+  im2show     = de_SelectImages(dset, nImages); %for vanhateren nImage = 16, but size(im2show) = 20?
 
   fig = de_NewFig('dataset-images', '__img', img_shape, nImages);
   set(fig.handle, 'Position', [0, 0, 1200, 1200]);
 
-  for ii=1:length(im2show)
+  for ii=1:min(4*4, length(im2show)) %make sure only 16 are shown
```
Comment (bcipolli) [inline code/stim/de_VisualizeDataset.m]: Once this constant gets duplicated, good to declare it somewhere and reuse as a constant.

**Review by bcipolli — CHANGES_REQUESTED** [review]
Overall, looks good! A couple of fine-tuning requests re: code / comments, but otherwise seems good to go!

File: code/analyze/ac/plots/de_PlotFFTs.m (hunk 3 above)
Comment (vishaalprasad) [reply code/analyze/ac/plots/de_PlotFFTs.m]: Well it's the pixel values that are normalized. But `Normalized power` works, I suppose?

**Comment by bcipolli** [comment]
Looks good. In the future, better to make PRs against `guruucsd/DifferentialEncoding`, which is the source repo. I will make a PR of these changes there.

**Pull request merged**